test/relations.test.js

```diff
@@ -5097,7 +5326,7 @@ describe('relations', function() {
 
     it('should find items on scope - findById', function(done) {
       Category.findOne(function(err, cat) {
-        cat.jobIds.should.eql([job2.id]);
+        cat.jobIds[0].should.eql(job2.id);
```

Comment:
This is almost the issue as I commented above - you are changing the assertion from "jobIds has a single item eql to job2.id" into "jobsIds has any number of items and the first one is eql to job2.id".

Please revert here and below too.